### PR TITLE
Bug 1144538 - [Settings] refactor TaskScheduler and CallConstant r=Arthur

### PR DESCRIPTION
**Pull request by EragonJ**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=972f9989f5b2)

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f9c5db9479ea)

File: apps/settings/js/panels/call/task_scheduler.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      this._isLocked = false;
+      this._executeNextTask();
+    },
+
+    /**
+     * We will filter out redundant tasks and execute only the remaining ones.
+     *
+     * @access private
+     * @memberOf TaskScheduler
+     * @param {Array} type
```
Comment (crh0716) [inline apps/settings/js/panels/call/task_scheduler.js]: `{String}`

File: apps/settings/js/panels/call/task_scheduler.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      }
+    },
+
+    /**
+     * This is the only entry point for caller, and we will enqueue all tasks
+     * one by one and execute them in order.
+     *
+     * @access public
+     * @memberOf TaskScheduler
+     * @param {Array} type
```
Comment (crh0716) [inline apps/settings/js/panels/call/task_scheduler.js]: `{String}`

**Comment by try-server-hook** [comment]
[EragonJ EragonJ (E.J.) (EragonJ)](https://github.com/EragonJ) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=444dad455e5e)

**Pull request merged**